core/modules/filter/filter.pages.inc

```diff
@@ -297,6 +301,19 @@ function filter_format_editor_image_form($form, &$form_state, $format) {
     '#access' => !empty($format->filters['filter_image_caption']->status),
   );
 
+  $form['library_view'] = array(
+    '#title' => t('Image library'),
+    '#type' => 'textfield',
+    '#default_value' => 'Image library',
+    '#wrapper_attributes' => array(
+      'class' => array('library-view'),
+      'data-editor-image-library-view' => views_embed_view('image_library'),
```

Comment:
I'm glad to see a data attribute used here to trigger the behavior, but embedding an entire view within a single HTML attribute is going to result in some awkward markup. This seems like it should be `#type = 'markup'`. Perhaps something like this:

```
$form['library_view'] = array(
  '#markup' => '<div class="library-view" data-editor-library-view="true">' . views_embed_view('image_library') . '</div>';  
);
```

Reply:
Not yet done.